cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/streaming/DefaultStreamingContext.java

```diff
@@ -48,6 +60,27 @@ public JavaSparkExecutionContext getSparkExecutionContext() {
   }
 
   @Override
+  public void registerLineage(final String referenceName) throws DatasetManagementException,
+    TransactionFailureException {
+    try {
+      if (!admin.datasetExists(referenceName)) {
+        admin.createDataset(referenceName, EXTERNAL_DATASET_TYPE, DatasetProperties.EMPTY);
+      }
+    } catch (InstanceConflictException ex) {
```

Comment:
should we have a jira to improve the `Admin`, so that it has a `createIfNotExists`, so if users use the `Admin`, they dont always have to put this sort of error handling in their code?